packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -27,6 +30,31 @@ @implementation RCTBaseTextInputView {
   BOOL _didMoveToWindow;
 }
 
+- (void)reactUpdateResponderOffsetForScrollView:(RCTScrollView *)scrollView
+{
+  if (![self isDescendantOfView:scrollView]) {
+    // View is outside scroll view
+    return;
+  }
+
+  UITextRange *selectedTextRange = self.backedTextInputView.selectedTextRange;
+  UITextSelectionRect *selection = [self.backedTextInputView selectionRectsForRange:selectedTextRange].firstObject;
+  if (selection == nil) {
+    // No active selection or caret - fallback to entire input frame
+    scrollView.firstResponderFocus = [self convertRect:self.frame toView:nil];
```

Comment:
`convertRect` is in the receiver's coordinate system, so this needs to be self.bounds